share/picongpu/tests/Synchrotron_plugin/lib/validate.py

```diff
@@ -0,0 +1,133 @@
+import sys
+import openpmd_api as io
+import numpy as np
+from synchrotron_lib import analytical_Propability, momentum_to_energy, quad, const
+
+
+# Used when comparing two sets of histogram data with different binning
+# given two set of vectors x0, y0 and x1, y1 we interpolate second set to the first set and subtract them
+# interpolate y1 to x0 and subtract y0 from y1
+def subtract_functions(x0, y0, x1, y1):
+    minX = max(np.min(x0), np.min(x1))
+    maxX = min(np.max(x0), np.max(x1))
+    #
+    x0 = [x for x in x0 if x >= minX and x <= maxX]
```

Comment:
@chillenzer what is the advantage of this suggested code?

Reply:
Well, not a strong one: It uses `numpy`'s masking, so it will be significantly faster (although that's likely not that much of a concern here). And to me personally
```python
x0 = x0[(x0 >= minX) * (x0 <= maxX)]
```
reads very nicely and concisely although that might be my bias having worked a lot with numpy. (I added the `np.asarray` later on because I wasn't sure if that function is supposed to take an array -- [type hints](https://peps.python.org/pep-0484/) would be a general improvement here.)